squire_web/src/tournament/overview/mod.rs

```diff
@@ -38,36 +36,54 @@ pub fn fetch_overview_data(ctx: &Context<TournOverview>, id: TournamentId) {
         TournOverviewMessage::OverviewQueryReady(data)
     })
 }
+*/
 
-impl Component for TournOverview {
-    type Message = TournOverviewMessage;
+impl TournViewerComponent for TournOverview {
     type Properties = OverviewProps;
+    type InteractionMessage = TournOverviewMessage;
+    type QueryMessage = TournOverviewQueryMessage;
 
-    fn create(ctx: &Context<Self>) -> Self {
-        let id = ctx.props().id;
-        fetch_overview_data(ctx, id);
+    fn v_create(_ctx: &Context<TournViewerComponentWrapper<Self>>, state: &WrapperState) -> Self {
+        let id = state.t_id.clone();
+        // fetch_overview_data(ctx, id);
         TournOverview { id, profile: None }
     }
 
-    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
-        match msg {
-            TournOverviewMessage::OverviewQueryReady(data) => {
+    fn load_queried_data(&mut self, _msg: Self::QueryMessage, _state: &WrapperState) -> bool {
+        match _msg {
+            TournOverviewQueryMessage::OverviewQueryReady(data) => {
                 let digest = self.profile != data || data.is_none();
                 self.profile = data;
                 digest
             }
         }
     }
 
-    fn view(&self, ctx: &Context<Self>) -> Html {
+    fn v_view(
+        &self,
+        _ctx: &Context<TournViewerComponentWrapper<Self>>,
+        _state: &WrapperState,
+    ) -> yew::Html {
         match self.profile.as_ref() {
             Some(p) => p.view(),
             None => {
-                fetch_overview_data(ctx, self.id);
+                // fetch_overview_data(_ctx, self.id);
                 Html::default()
             }
         }
     }
+
+    fn query(
+        &mut self,
+        _ctx: &Context<TournViewerComponentWrapper<Self>>,
+        _state: &WrapperState,
+    ) -> Box<dyn 'static + Send + FnOnce(&TournamentManager) -> Self::QueryMessage> {
```

Comment:
Did we end up making a type alias for this? If not, we should.
```suggestion
    ) -> Query<Self::QueryMessage> {
```